src/domain/sales-channels/tables/product.tsx

```diff
@@ -0,0 +1,462 @@
+      if (query) {
+        setFreeText(query)
+        gotoPage(0)
+      } else {
+        if (typeof query !== "undefined") {
+          // if we delete query string, we reset the table view
+          reset()
+        }
+      }
+    }, 400)
+
+    return () => clearTimeout(delayDebounceFn)
+  }, [query])
+
+  const handleNext = () => {
+    if (canNextPage) {
+      paginate(1)
+      nextPage()
+    }
+  }
+
+  const handlePrev = () => {
+    if (canPreviousPage) {
+      paginate(-1)
+      previousPage()
+    }
+  }
+
+  const getActions = (id: string) => [
+    {
+      label: "Details",
+      onClick: () => navigate(`/a/products/${id}`),
+      icon: <DetailsIcon size={20} />,
+    },
+    {
+      label: "Remove from the channel",
+      variant: "danger",
+      onClick: () => removeProductFromSalesChannel(id),
+      icon: <TrashIcon size={20} />,
+    },
+  ]
+
+  return (
+    <div className="w-full h-[880px] overflow-y-auto flex flex-col">
+      <Table
+        containerClassName="flex-1"
+        filteringOptions={
+          filters && (
+            <ProductsFilter
+              filters={filters}
+              submitFilters={setFilters}
+              clearFilters={clearFilters}
+              tabs={filterTabs}
+              onTabClick={setTab}
+              activeTab={activeFilterTab}
+              onRemoveTab={removeTab}
+              onSaveTab={saveTab}
+            />
+          )
+        }
+        enableSearch
+        handleSearch={setQuery}
+        {...getTableProps()}
+      >
+        <Table.Head>
+          {headerGroups?.map((headerGroup) => (
+            <Table.HeadRow {...headerGroup.getHeaderGroupProps()}>
+              {headerGroup.headers.map((col) => (
+                <Table.HeadCell {...col.getHeaderProps()}>
+                  {col.render("Header")}
+                </Table.HeadCell>
+              ))}
+            </Table.HeadRow>
+          ))}
+        </Table.Head>
+        <Table.Body {...getTableBodyProps()}>
+          {rows.map((row) => {
+            prepareRow(row)
+            return (
+              <ProductRow
+                row={row}
+                actions={!isAddTable ? getActions(row.original.id) : undefined}
+              />
+            )
+          })}
+        </Table.Body>
+      </Table>
+      <TablePagination
+        count={count!}
+        limit={limit}
+        offset={offs}
+        pageSize={offs + rows.length}
+        title="Products"
+        currentPage={pageIndex + 1}
+        pageCount={pageCount}
+        nextPage={handleNext}
+        prevPage={handlePrev}
+        hasNext={canNextPage}
+        hasPrev={canPreviousPage}
+      />
+    </div>
+  )
+}
+
+/**
+ * Renders product table row.
+ */
+const ProductRow = ({ row, actions }) => {
+  const product = row.original
+
+  return (
+    <Table.Row
+      color={"inherit"}
+      className={row.isSelected ? "bg-grey-5" : ""}
+      linkTo={`/a/products/${product.id}`}
```

Comment:
Suggestion(non-blocking): It might be fine to not link to the product on each row. As you mentioned it's really annoying if you misclick the checkbox. We could consider selecting the row instead.

Reply:
agree, especially since there is link to the details page in row menu

Reply:
we should remove the linkTo property and update the table.row component to reflect this